homeassistant/components/hydrawise/sensor.py

```diff
@@ -63,15 +120,45 @@ class HydrawiseSensor(HydrawiseEntity, SensorEntity):
 
     def _update_attrs(self) -> None:
         """Update state attributes."""
-        if self.entity_description.key == "watering_time":
-            if (current_run := self.zone.scheduled_runs.current_run) is not None:
-                self._attr_native_value = int(
-                    current_run.remaining_time.total_seconds() / 60
-                )
-            else:
-                self._attr_native_value = 0
-        elif self.entity_description.key == "next_cycle":
-            if (next_run := self.zone.scheduled_runs.next_run) is not None:
-                self._attr_native_value = dt_util.as_utc(next_run.start_time)
-            else:
-                self._attr_native_value = datetime.max.replace(tzinfo=dt_util.UTC)
+        specific_getter = getattr(self, f"_get_{self.entity_description.key}")
+        value = specific_getter()
+        if value is not None:
+            self._attr_native_value = value
+
+    def _get_watering_time(self) -> int:
+        if (current_run := self.zone.scheduled_runs.current_run) is not None:
+            return int(current_run.remaining_time.total_seconds() / 60)
+        return 0
+
+    def _get_next_cycle(self) -> datetime:
+        if (next_run := self.zone.scheduled_runs.next_run) is not None:
+            return dt_util.as_utc(next_run.start_time)
+        return datetime.max.replace(tzinfo=dt_util.UTC)
+
+    def _get_daily_active_water_use(self) -> Any:
+        daily_water_summary = self.coordinator.data.daily_water_use[self.controller.id]
+        if self.zone is None and self.sensor is not None:
+            # water use for the controller
+            return daily_water_summary.total_active_use
+        if self.zone is not None:
```

Comment:
If you put this condition first, then the controller water use only needs to check for `self.sensor is not None`.

Or are there sensors that are associated with zones that we'll want to handle later?

Reply:
Fixed